aibolit/model/stats.py

```diff
@@ -0,0 +1,208 @@
+
+    def count_acts(self, acts, ranked):
+        patterns_numbers = ranked[:, 0]
+        # number of times when pattern was on first place,
+        # if we decrease pattern by 1/ncss
+        m = np.zeros(ranked.shape[1])
+        # number of times when pattern was on first place,
+        # if we increase pattern by 1/ncss
+        p = np.zeros(ranked.shape[1])
+        for i in range(len(patterns_numbers)):
+            if acts[i] == 1:
+                m[patterns_numbers[i]] += 1
+            elif acts[i] == 2:
+                p[patterns_numbers[i]] += 1
+        return m, p
+
+    def get_patterns_name(self):
+        only_patterns = []
+        patterns_code = []
+        config = Config.get_patterns_config()
+        for x in config['patterns']:
+            if x['code'] not in config['patterns_exclude']:
+                only_patterns.append(x['name'])
+                patterns_code.append(x['code'])
+        features_number = len(only_patterns)
+        print("Number of features: ", features_number)
+        patterns = {x['code']: x['name'] for x in config['patterns']}
+        metrics = {x['code']: x['name'] for x in config['metrics']}
+        replace_dict = dict(patterns, **metrics)
+        return replace_dict
+
+    def get_table(
+            self,
+            features_conf: Dict[Any, Any],
+            m,
+            p,
+            acts_complexity) -> pd.DataFrame:
+        """
+        Prints results, given with `check_impact`.
+
+
+        :param features_conf: features config of model
+        :param m: number of times when pattern was on first place,
+        if we decrease pattern by 1/ncss
+        :param p: number of times when pattern was on first place,
+        if we increase pattern by 1/ncss
+        :param acts_complexity:
+
+        """
+
+        df = pd.DataFrame(columns=[
+            'pattern', ' -1(top1)', '+1(top1)',
+            'p-c-', 'p+c+', 'p-c+', 'p+c-', 'p-c=',
+            'p+c='])
+        replace_dict = self.get_patterns_name()
+        for i in range(len(features_conf)):
+            top_minus = int(m[i])
+            top_plus = int(p[i])
+            p_minus_c_minus = int(acts_complexity[i, 0])
+            p_plus_c_plus = int(acts_complexity[i, 4])
+            p_minus_c_plus = int(acts_complexity[i, 1])
+            p_plus_c_minus = int(acts_complexity[i, 3])
+            p_minus_c_euq = int(acts_complexity[i, 2])
+            p_plus_c_euq = int(acts_complexity[i, 5])
+            pattern = replace_dict.get(features_conf[i])
+            df = df.append({
+                'pattern': pattern, ' -1(top1)': top_minus, '+1(top1)': top_plus,
+                'p-c-': p_minus_c_minus, 'p+c+': p_plus_c_plus, 'p-c+': p_minus_c_plus,
+                'p+c-': p_plus_c_minus, 'p-c=': p_minus_c_euq,
+                'p+c=': p_plus_c_euq
+            }, ignore_index=True)
+
+        return df
+
+    def divide_array(self, X, pattern_idx):
```

Comment:
Better with mask. Return X[X[:, pattern_idx] == 0,:], X[X[:, pattern_idx] > 0,:]

Reply:
we divide here only using zero mask, by definition